docs/jenkins-setup.txt

```diff
@@ -0,0 +1,161 @@
+		(from `stella hijinks`)
+	- netmask: 255.255.254.0
+		(from `moira 5 1 sipb-460`)
+	- gateway: 18.4.60.1
+		(from `moira 5 1 sipb-460`, but note that
+			the last octet should be 1, by convention
+			for network gateways)
+	- nameservers: 18.72.0.3 18.70.0.160 18.71.0.151
+		(from `dig mit.edu` from within MIT's network)
+
+=== Standard setup ===
+apt update
+apt upgrade
+apt dist-upgrade
+apt install sudo openssh-server git
+apt install vim htop less xclip
+ssh-keygen -lf /etc/ssh/ssh_host_ecdsa_key.pub
+usermod -aG sudo <user>
+
+ssh-copy-id <user>@18.4.60.253
+ssh <user>@18.4.60.253
+# disable password authentication in /etc/ssh/sshd_config
+
+=== Docker ===
+sudo apt install apt-transport-https ca-certificates curl gnupg2 software-properties-common
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
+# check fingerprint
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+sudo apt update && sudo apt install docker-ce
+
+=== nginx ===
+sudo apt install nginx
+sudo cp /etc/nginx/sites-available/default /etc/nginx/sites-available/hijinks
+sudo ln -s /etc/nginx/sites-available/hijinks /etc/nginx/sites-enabled/hijinks
+sudo rm /etc/nginx/sites-enabled/default
+# edit /etc/nginx/sites-available/hijinks:
+	server_name hijinks.mit.edu;
+sudo rm /var/www/html/index.nginx-debian.html
+
+=== Let's Encrypt ===
+sudo bash -c "echo 'deb http://ftp.debian.org/debian stretch-backports main' >> /etc/apt/sources.list"
+sudo apt update
+sudo apt install python-certbot-nginx -t stretch-backports
+sudo certbot --nginx --must-staple --hsts
+	- yes to redirect
+sudo certbot renew --dry-run
+# edit /etc/nginx/sites-available/hijinks: (replace location / block)
+	location /jenkins/ {
+		include /etc/nginx/proxy_params;
+		proxy_pass http://localhost:8080;
+		proxy_read_timeout 90s;
+		proxy_redirect http://localhost:8080 https://hijinks.mit.edu;
+		sendfile off;
+		if ($scheme != "https") {
+			return 301 https://$host$request_uri;
+		}
+	}
+sudo nginx -t && sudo nginx -s reload
+
+=== Jenkins ===
+sudo apt install default-jre
+wget -q -O - https://pkg.jenkins.io/debian/jenkins.io.key | sudo apt-key add -
+# check fingerprint
+sudo sh -c 'echo deb http://pkg.jenkins.io/debian-stable binary/ > /etc/apt/sources.list.d/jenkins.list'
+sudo apt update && sudo apt install jenkins
+# add "--prefix=/jenkins" in JENKINS_ARGS in /etc/default/jenkins
+sudo usermod -aG docker jenkins
+systemctl restart jenkins
+sudo cat /var/lib/jenkins/secrets/initialAdminPassword
+# enter password into https://hijinks.mit.edu/jenkins/
+# select plugins:
+	- Build Timeout
+	- Timestamper
+	- Pipeline
+	- Github Branch Source
+	- Pipeline: Stage View
+	- Git
+	- Github
+# create user
+# jenkins url: https://hijinks.mit.edu/jenkins/
+systemctl restart jenkins # jenkins bug
+
+=== Jenkins Homeworld ===
+# Manage Jenkins > Configure System > set # of executors 1
+	(protection in case locks don't work)
```

Comment:
Do locks work? (Can we change this?)

Reply:
Locks should work, but there currently isn't any reason to have more than one executor, so I just left the setting as is.

I could reword/remove the comment though.